pkg/controller/plan/migration.go

```diff
@@ -1502,22 +1520,16 @@ func (r *Migration) updateCopyProgress(vm *plan.VMStatus, step *plan.Step) (err
 			conditions := dv.Conditions()
 			switch dv.Status.Phase {
 			case cdi.Succeeded, cdi.Paused:
-				completed++
-				task.Phase = Completed
-				task.Reason = TransferCompleted
-				task.Progress.Completed = task.Progress.Total
-				task.MarkCompleted()
-			case cdi.Pending, cdi.ImportScheduled:
-				pending++
-				task.Phase = Pending
-				cnd := conditions.FindCondition("Bound")
-				if cnd != nil && cnd.Status == True {
-					cnd = conditions.FindCondition("Running")
+				r.setTaskCompleted(task, completed)
+			case cdi.PendingPopulation:
+				switch r.Source.Provider.Type() {
+				case v1beta1.VSphere:
+					r.setTaskCompleted(task, completed)
+				default:
+					pendingReason = r.setTaskPending(task, pending, conditions)
```

Comment:
wouldn't it be simpler to replace PendingPopulation with Succeeded in case the source provider is of type vSphere before getting here? something like:
```golang
if dv.Status.Phase == cdi.PendingPopulation && r.Source.Provider.Type() == v1beta1.VSphere {
  dv.Status.Phase = cdi.Succeeded
}
```